### PR TITLE
initial pr

### PR DESCRIPTION
**Pull request by prerana1821**

### CODE REVIEW

File: data.js [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
 faker.seed(123);
 
+// // Products
+// const categories = [
+//   "Men's Swim Shorts",
+//   "Floatation Devices",
+//   "Training Kickboards",
+//   "Kiddie Pools",
+//   "Swimming Kits",
+// ];
```
Comment (AnkitaBagale) [inline data.js]: remove commented code

File: middlewares/auth-verify.middleware.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const jwt = require("jsonwebtoken");
+const jwtSecret = process.env['jwt-secret'];
```
Comment (AnkitaBagale) [inline middlewares/auth-verify.middleware.js]: Just a suggestion, it's a convention to have env variable names in capital case.

File: routes/auth.router.js [hunk 3]
```diff
@@ -1,27 +1,36 @@
...
 const { Auth } = require("../models/auth.model");
 const { User } = require("../models/user.model");
 
+const jwtSecret = process.env['jwt-secret'];
+
 const findUserByUserName = (username) => {
   return Auth.findOne({ username: new RegExp('^' + username + '$', "i") }, function(err, user) {
     if (err) return console.log(err);
   })
 };
```
Comment (AnkitaBagale) [inline routes/auth.router.js]: Looks like a utility function, we can move it to util folder

File: routes/auth.router.js [hunk 4]
```diff
@@ -1,27 +1,36 @@
...
 
-router.get("/", async (req, res) => {
-  const auth = await Auth.find({});
-  res.json({ auth, success: true })
-})
+const generateToken = (userId) => {
+  return jwt.sign({ userId }, jwtSecret, { expiresIn: '24h' });
+}
```
Comment (AnkitaBagale) [inline routes/auth.router.js]: Looks like a utility function, we can move it to util folder

File: routes/auth.router.js [hunk 5]
```diff
@@ -1,27 +1,36 @@
...
-      return res.status(200).json({ user, success: true, message: "Login Successful" })
-    } res.status(403).json({ success: false, message: "Wrong Password. Enter correct password" })
-  } res.status(404).json({ success: false, message: "User not found. Check your user credentials" })
+  try {
+    const user = await findUserByUserName(username);
+    if (user) {
+      const validPassword = await bcrypt.compare(password, user.password);
+      if (validPassword) {
+        const token = generateToken(user._id);
+        return res.status(200).json({ user, token, success: true, message: "Login Successful" })
```
Comment (AnkitaBagale) [inline routes/auth.router.js]: Try not to nest if statements
instead write as below, I feel it more readable
```
if(!user){ return res with 404}
if(!validPassword) {return res with 403}
return res with 200
```

File: routes/user.router.js [hunk 6]
```diff
@@ -1,157 +1,247 @@
...
+      const { userId } = req.user;
+      const user = await User.findById(userId);
+      if (!user) {
+        return res.status(400).json({ success: false, errorMessage: "unable to find user" });
+      } else {
+        const wholeObj = await user.populate('cart.productId').execPopulate();
+        const object = wholeObj.cart.map((item) => {
+          const { _id, productId, quantity } = item;
+          return { _id, productId: { ...productId._doc, quantity } }
+        })
```
Comment (AnkitaBagale) [inline routes/user.router.js]: we can create util function to make it cleaner. and i also i gues this is repeating

**Comment by AnkitaBagale** [comment]
Hi Prerana,
I have completed your PR review, Very well done! I just have 2 suggestions
1.Try not to nest if statements
instead, write as below, I feel it more readable
```
if(!user){ return res with 404}
if(!validPassword) {return res with 403}
return res with 200
```
2. Create small utility functions for DRY and keep them in util folder

**Comment by prerana1821** [comment]
> Hi Prerana,
> I have completed your PR review, Very well done! I just have 2 suggestions
> 1.Try not to nest if statements
> instead, write as below, I feel it more readable
> 
> ```
> if(!user){ return res with 404}
> if(!validPassword) {return res with 403}
> return res with 200
> ```
> 
> 1. Create small utility functions for DRY and keep them in util folder

Hey Ankita, thank you for reviewing my code.
I have made all the suggested changes! Thanks again.

**Pull request closed without merging**